tools/composefs-from-json.c

```diff
@@ -536,20 +541,20 @@ static int fill_file(const char *typ, struct lcfs_node_s *root,
 		cleanup_free uint8_t *buf = malloc(buf_size);
 
 		if (buf == NULL) {
-			error(0, 0, "malloc");
+			fprintf(stderr, "malloc");
```

Comment:
This is missing a newline though.  Why are we printing here?  Supporting custom strings really requires going to a whole GError/Result style error handling; in baseline C we can only set errno right?

Reply:
Well, its a tool, not a library, so while not ideal, printing here isn't completely wrong. It should probably exit on printing tho.

Reply:
I added the newline, and we do generally exit after the print, the tool just prints two error lines.

I think this is fine for now, because composefs-from-json really is an internal thing at this point.